MueLu: link failure with RDC enabled due to multiple libraries including the same object file

@trilinos/muelu 

MueLu does not successfully remove object files which are supposed to be only in the libmuelu-interface.a are also appearing in libmuelu.a. Below is a printout from CMake printing out the SOURCES, SOURCES_INTERFACE and SOURCES after SOURCES_INTERFACE are supposedly removed. Looks like SOURCES_INTERFACE has some double slashes in its names which make it not pattern match, and thus leaving the stuff in the link list for the libraries. While this appears to be normally not a problem because compilers are nice to people who misuse the link system, nvcc will complain if relocatable device code is on. I assume at that point nvcc runs the device linker, and checks that there are no duplicated symbols in the link lines. 

Removing the double slashes in the muelu/src/CmakeLists.txt solves the issue, but someone (Chris ;-) ) left comments in there in 2015 that it is critical to have. Maybe this is something which got fixed with later CMake versions? 

CMake print out:
```
SOURCES = Utils/ExplicitInstantiation/MueLu_AggregationExportFactory.cpp;Utils/ExplicitInstantiation/MueLu_AmalgamationFactory.cpp;Utils/ExplicitInstantiation/MueLu_Amesos2Smoother.cpp;Utils/ExplicitInstantiation/MueLu_AMGXOperator.cpp;Utils/ExplicitInstantiation/MueLu_AlgebraicPermutationStrategy.cpp;Utils/ExplicitInstantiation/MueLu_BlackBoxPFactory.cpp;Utils/ExplicitInstantiation/MueLu_BlockedCoarseMapFactory.cpp;Utils/ExplicitInstantiation/MueLu_BlockedDirectSolver.cpp;Utils/ExplicitInstantiation/MueLu_BlockedGaussSeidelSmoother.cpp;Utils/ExplicitInstantiation/MueLu_BlockedJacobiSmoother.cpp;Utils/ExplicitInstantiation/MueLu_BlockedPFactory.cpp;Utils/ExplicitInstantiation/MueLu_BlockedRAPFactory.cpp;Utils/ExplicitInstantiation/MueLu_BrickAggregationFactory.cpp;Utils/ExplicitInstantiation/MueLu_BraessSarazinSmoother.cpp;Utils/ExplicitInstantiation/MueLu_CGSolver.cpp;Utils/ExplicitInstantiation/MueLu_CloneRepartitionInterface.cpp;Utils/ExplicitInstantiation/MueLu_CoalesceDropFactory.cpp;Utils/ExplicitInstantiation/MueLu_CoalesceDropFactory_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_CoarseMapFactory.cpp;Utils/ExplicitInstantiation/MueLu_CoarseMapFactory_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_CoarseningVisualizationFactory.cpp;Utils/ExplicitInstantiation/MueLu_Constraint.cpp;Utils/ExplicitInstantiation/MueLu_ConstraintFactory.cpp;Utils/ExplicitInstantiation/MueLu_CoordinatesTransferFactory.cpp;Utils/ExplicitInstantiation/MueLu_CoordinatesTransferFactory_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_CoupledRBMFactory.cpp;Utils/ExplicitInstantiation/MueLu_DemoFactory.cpp;Utils/ExplicitInstantiation/MueLu_DirectSolver.cpp;Utils/ExplicitInstantiation/MueLu_DropNegativeEntriesFactory.cpp;Utils/ExplicitInstantiation/MueLu_EminPFactory.cpp;Utils/ExplicitInstantiation/MueLu_FacadeClassFactory.cpp;Utils/ExplicitInstantiation/MueLu_FactoryManager.cpp;Utils/ExplicitInstantiation/MueLu_FilteredAFactory.cpp;Utils/ExplicitInstantiation/MueLu_FineLevelInputDataFactory.cpp;Utils/ExplicitInstantiation/MueLu_GeneralGeometricPFactory.cpp;Utils/ExplicitInstantiation/MueLu_GenericRFactory.cpp;Utils/ExplicitInstantiation/MueLu_GMRESSolver.cpp;Utils/ExplicitInstantiation/MueLu_Hierarchy.cpp;Utils/ExplicitInstantiation/MueLu_HierarchyUtils.cpp;Utils/ExplicitInstantiation/MueLu_Ifpack2Smoother.cpp;Utils/ExplicitInstantiation/MueLu_IndefBlockedDiagonalSmoother.cpp;Utils/ExplicitInstantiation/MueLu_IntrepidPCoarsenFactory.cpp;Utils/ExplicitInstantiation/MueLu_LineDetectionFactory.cpp;Utils/ExplicitInstantiation/MueLu_LocalPermutationStrategy.cpp;Utils/ExplicitInstantiation/MueLu_MapTransferFactory.cpp;Utils/ExplicitInstantiation/MueLu_MatrixAnalysisFactory.cpp;Utils/ExplicitInstantiation/MueLu_MergedBlockedMatrixFactory.cpp;Utils/ExplicitInstantiation/MueLu_MergedSmoother.cpp;Utils/ExplicitInstantiation/MueLu_MultiVectorTransferFactory.cpp;Utils/ExplicitInstantiation/MueLu_NullspaceFactory.cpp;Utils/ExplicitInstantiation/MueLu_NullspaceFactory_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_NullspacePresmoothFactory.cpp;Utils/ExplicitInstantiation/MueLu_PatternFactory.cpp;Utils/ExplicitInstantiation/MueLu_PerfUtils.cpp;Utils/ExplicitInstantiation/MueLu_PermutationFactory.cpp;Utils/ExplicitInstantiation/MueLu_PermutingSmoother.cpp;Utils/ExplicitInstantiation/MueLu_PgPFactory.cpp;Utils/ExplicitInstantiation/MueLu_PreDropFunctionConstVal.cpp;Utils/ExplicitInstantiation/MueLu_ProjectorSmoother.cpp;Utils/ExplicitInstantiation/MueLu_RAPFactory.cpp;Utils/ExplicitInstantiation/MueLu_RAPShiftFactory.cpp;Utils/ExplicitInstantiation/MueLu_RebalanceAcFactory.cpp;Utils/ExplicitInstantiation/MueLu_RebalanceBlockAcFactory.cpp;Utils/ExplicitInstantiation/MueLu_RebalanceBlockInterpolationFactory.cpp;Utils/ExplicitInstantiation/MueLu_RebalanceBlockRestrictionFactory.cpp;Utils/ExplicitInstantiation/MueLu_RebalanceTransferFactory.cpp;Utils/ExplicitInstantiation/MueLu_ReorderBlockAFactory.cpp;Utils/ExplicitInstantiation/MueLu_RepartitionFactory.cpp;Utils/ExplicitInstantiation/MueLu_RepartitionHeuristicFactory.cpp;Utils/ExplicitInstantiation/MueLu_RigidBodyModeFactory.cpp;Utils/ExplicitInstantiation/MueLu_SaPFactory.cpp;Utils/ExplicitInstantiation/MueLu_SaPFactory_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_SchurComplementFactory.cpp;Utils/ExplicitInstantiation/MueLu_SegregatedAFactory.cpp;Utils/ExplicitInstantiation/MueLu_ShiftedLaplacian.cpp;Utils/ExplicitInstantiation/MueLu_ShiftedLaplacianOperator.cpp;Utils/ExplicitInstantiation/MueLu_SimpleSmoother.cpp;Utils/ExplicitInstantiation/MueLu_SmootherFactory.cpp;Utils/ExplicitInstantiation/MueLu_SmootherPrototype.cpp;Utils/ExplicitInstantiation/MueLu_SteepestDescentSolver.cpp;Utils/ExplicitInstantiation/MueLu_StructuredLineDetectionFactory.cpp;Utils/ExplicitInstantiation/MueLu_SubBlockAFactory.cpp;Utils/ExplicitInstantiation/MueLu_TekoSmoother.cpp;Utils/ExplicitInstantiation/MueLu_TentativePFactory.cpp;Utils/ExplicitInstantiation/MueLu_TentativePFactory_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_ThresholdAFilterFactory.cpp;Utils/ExplicitInstantiation/MueLu_ToggleCoordinatesTransferFactory.cpp;Utils/ExplicitInstantiation/MueLu_TogglePFactory.cpp;Utils/ExplicitInstantiation/MueLu_TopRAPFactory.cpp;Utils/ExplicitInstantiation/MueLu_TopSmootherFactory.cpp;Utils/ExplicitInstantiation/MueLu_TpetraOperator.cpp;Utils/ExplicitInstantiation/MueLu_TransPFactory.cpp;Utils/ExplicitInstantiation/MueLu_TrilinosSmoother.cpp;Utils/ExplicitInstantiation/MueLu_UnsmooshFactory.cpp;Utils/ExplicitInstantiation/MueLu_UserPFactory.cpp;Utils/ExplicitInstantiation/MueLu_Utilities.cpp;Utils/ExplicitInstantiation/MueLu_UtilitiesBase.cpp;Utils/ExplicitInstantiation/MueLu_Utilities_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_VariableDofLaplacianFactory.cpp;Utils/ExplicitInstantiation/MueLu_SemiCoarsenPFactory.cpp;Utils/ExplicitInstantiation/MueLu_UzawaSmoother.cpp;Utils/ExplicitInstantiation/MueLu_VisualizationHelpers.cpp;Utils/ExplicitInstantiation/MueLu_ZoltanInterface.cpp;Utils/ExplicitInstantiation/MueLu_Zoltan2Interface.cpp;Utils/ExplicitInstantiation/MueLu_Aggregates.cpp;Utils/ExplicitInstantiation/MueLu_Aggregates_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase1Algorithm.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase1Algorithm_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase2aAlgorithm.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase2aAlgorithm_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase2bAlgorithm.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase2bAlgorithm_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase3Algorithm.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase3Algorithm_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_AmalgamationInfo.cpp;Utils/ExplicitInstantiation/MueLu_CoupledAggregationCommHelper.cpp;Utils/ExplicitInstantiation/MueLu_CoupledAggregationFactory.cpp;Utils/ExplicitInstantiation/MueLu_Graph.cpp;Utils/ExplicitInstantiation/MueLu_IsolatedNodeAggregationAlgorithm.cpp;Utils/ExplicitInstantiation/MueLu_IsolatedNodeAggregationAlgorithm_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_IsorropiaInterface.cpp;Utils/ExplicitInstantiation/MueLu_LWGraph.cpp;Utils/ExplicitInstantiation/MueLu_LWGraph_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_LeftoverAggregationAlgorithm.cpp;Utils/ExplicitInstantiation/MueLu_LocalAggregationAlgorithm.cpp;Utils/ExplicitInstantiation/MueLu_OnePtAggregationAlgorithm.cpp;Utils/ExplicitInstantiation/MueLu_OnePtAggregationAlgorithm_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_PreserveDirichletAggregationAlgorithm.cpp;Utils/ExplicitInstantiation/MueLu_PreserveDirichletAggregationAlgorithm_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_RebalanceMapFactory.cpp;Utils/ExplicitInstantiation/MueLu_RepartitionInterface.cpp;Utils/ExplicitInstantiation/MueLu_UncoupledAggregationFactory.cpp;Utils/ExplicitInstantiation/MueLu_UncoupledAggregationFactory_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_UserAggregationFactory.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Graph/MueLu_LinkedList.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListInterpreter.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListUtils.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_Describable.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_Factory.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_FactoryBase.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_Level.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_MasterList.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_NoFactory.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_VerboseObject.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_VerbosityLevel.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Smoothers/MueLu_AmesosSmoother.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Smoothers/MueLu_IfpackSmoother.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/MueLu_Memory.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/MueLu_Monitor.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/MueLu_MutuallyExclusiveTime.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/MueLu_Utilities.cpp
SOURCES_INTERFACE = /mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface//MueLu_ML2MueLuParameterTranslator.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface//MueLu_ParameterListInterpreter.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface//MueLu_ParameterListUtils.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/ExplicitInstantiation/MueLu_FacadeClassFactory.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/ExplicitInstantiation/MueLu_ParameterListInterpreter.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/ExplicitInstantiation/MueLu_MLParameterListInterpreter.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/ExplicitInstantiation/MueLu_FacadeSimple.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/ExplicitInstantiation/MueLu_FacadeBGS2x2.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/ExplicitInstantiation/MueLu_FactoryFactory.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/ExplicitInstantiation/MueLu_AdaptiveSaMLParameterListInterpreter.cpp
SOURCES = Utils/ExplicitInstantiation/MueLu_AggregationExportFactory.cpp;Utils/ExplicitInstantiation/MueLu_AmalgamationFactory.cpp;Utils/ExplicitInstantiation/MueLu_Amesos2Smoother.cpp;Utils/ExplicitInstantiation/MueLu_AMGXOperator.cpp;Utils/ExplicitInstantiation/MueLu_AlgebraicPermutationStrategy.cpp;Utils/ExplicitInstantiation/MueLu_BlackBoxPFactory.cpp;Utils/ExplicitInstantiation/MueLu_BlockedCoarseMapFactory.cpp;Utils/ExplicitInstantiation/MueLu_BlockedDirectSolver.cpp;Utils/ExplicitInstantiation/MueLu_BlockedGaussSeidelSmoother.cpp;Utils/ExplicitInstantiation/MueLu_BlockedJacobiSmoother.cpp;Utils/ExplicitInstantiation/MueLu_BlockedPFactory.cpp;Utils/ExplicitInstantiation/MueLu_BlockedRAPFactory.cpp;Utils/ExplicitInstantiation/MueLu_BrickAggregationFactory.cpp;Utils/ExplicitInstantiation/MueLu_BraessSarazinSmoother.cpp;Utils/ExplicitInstantiation/MueLu_CGSolver.cpp;Utils/ExplicitInstantiation/MueLu_CloneRepartitionInterface.cpp;Utils/ExplicitInstantiation/MueLu_CoalesceDropFactory.cpp;Utils/ExplicitInstantiation/MueLu_CoalesceDropFactory_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_CoarseMapFactory.cpp;Utils/ExplicitInstantiation/MueLu_CoarseMapFactory_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_CoarseningVisualizationFactory.cpp;Utils/ExplicitInstantiation/MueLu_Constraint.cpp;Utils/ExplicitInstantiation/MueLu_ConstraintFactory.cpp;Utils/ExplicitInstantiation/MueLu_CoordinatesTransferFactory.cpp;Utils/ExplicitInstantiation/MueLu_CoordinatesTransferFactory_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_CoupledRBMFactory.cpp;Utils/ExplicitInstantiation/MueLu_DemoFactory.cpp;Utils/ExplicitInstantiation/MueLu_DirectSolver.cpp;Utils/ExplicitInstantiation/MueLu_DropNegativeEntriesFactory.cpp;Utils/ExplicitInstantiation/MueLu_EminPFactory.cpp;Utils/ExplicitInstantiation/MueLu_FacadeClassFactory.cpp;Utils/ExplicitInstantiation/MueLu_FactoryManager.cpp;Utils/ExplicitInstantiation/MueLu_FilteredAFactory.cpp;Utils/ExplicitInstantiation/MueLu_FineLevelInputDataFactory.cpp;Utils/ExplicitInstantiation/MueLu_GeneralGeometricPFactory.cpp;Utils/ExplicitInstantiation/MueLu_GenericRFactory.cpp;Utils/ExplicitInstantiation/MueLu_GMRESSolver.cpp;Utils/ExplicitInstantiation/MueLu_Hierarchy.cpp;Utils/ExplicitInstantiation/MueLu_HierarchyUtils.cpp;Utils/ExplicitInstantiation/MueLu_Ifpack2Smoother.cpp;Utils/ExplicitInstantiation/MueLu_IndefBlockedDiagonalSmoother.cpp;Utils/ExplicitInstantiation/MueLu_IntrepidPCoarsenFactory.cpp;Utils/ExplicitInstantiation/MueLu_LineDetectionFactory.cpp;Utils/ExplicitInstantiation/MueLu_LocalPermutationStrategy.cpp;Utils/ExplicitInstantiation/MueLu_MapTransferFactory.cpp;Utils/ExplicitInstantiation/MueLu_MatrixAnalysisFactory.cpp;Utils/ExplicitInstantiation/MueLu_MergedBlockedMatrixFactory.cpp;Utils/ExplicitInstantiation/MueLu_MergedSmoother.cpp;Utils/ExplicitInstantiation/MueLu_MultiVectorTransferFactory.cpp;Utils/ExplicitInstantiation/MueLu_NullspaceFactory.cpp;Utils/ExplicitInstantiation/MueLu_NullspaceFactory_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_NullspacePresmoothFactory.cpp;Utils/ExplicitInstantiation/MueLu_PatternFactory.cpp;Utils/ExplicitInstantiation/MueLu_PerfUtils.cpp;Utils/ExplicitInstantiation/MueLu_PermutationFactory.cpp;Utils/ExplicitInstantiation/MueLu_PermutingSmoother.cpp;Utils/ExplicitInstantiation/MueLu_PgPFactory.cpp;Utils/ExplicitInstantiation/MueLu_PreDropFunctionConstVal.cpp;Utils/ExplicitInstantiation/MueLu_ProjectorSmoother.cpp;Utils/ExplicitInstantiation/MueLu_RAPFactory.cpp;Utils/ExplicitInstantiation/MueLu_RAPShiftFactory.cpp;Utils/ExplicitInstantiation/MueLu_RebalanceAcFactory.cpp;Utils/ExplicitInstantiation/MueLu_RebalanceBlockAcFactory.cpp;Utils/ExplicitInstantiation/MueLu_RebalanceBlockInterpolationFactory.cpp;Utils/ExplicitInstantiation/MueLu_RebalanceBlockRestrictionFactory.cpp;Utils/ExplicitInstantiation/MueLu_RebalanceTransferFactory.cpp;Utils/ExplicitInstantiation/MueLu_ReorderBlockAFactory.cpp;Utils/ExplicitInstantiation/MueLu_RepartitionFactory.cpp;Utils/ExplicitInstantiation/MueLu_RepartitionHeuristicFactory.cpp;Utils/ExplicitInstantiation/MueLu_RigidBodyModeFactory.cpp;Utils/ExplicitInstantiation/MueLu_SaPFactory.cpp;Utils/ExplicitInstantiation/MueLu_SaPFactory_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_SchurComplementFactory.cpp;Utils/ExplicitInstantiation/MueLu_SegregatedAFactory.cpp;Utils/ExplicitInstantiation/MueLu_ShiftedLaplacian.cpp;Utils/ExplicitInstantiation/MueLu_ShiftedLaplacianOperator.cpp;Utils/ExplicitInstantiation/MueLu_SimpleSmoother.cpp;Utils/ExplicitInstantiation/MueLu_SmootherFactory.cpp;Utils/ExplicitInstantiation/MueLu_SmootherPrototype.cpp;Utils/ExplicitInstantiation/MueLu_SteepestDescentSolver.cpp;Utils/ExplicitInstantiation/MueLu_StructuredLineDetectionFactory.cpp;Utils/ExplicitInstantiation/MueLu_SubBlockAFactory.cpp;Utils/ExplicitInstantiation/MueLu_TekoSmoother.cpp;Utils/ExplicitInstantiation/MueLu_TentativePFactory.cpp;Utils/ExplicitInstantiation/MueLu_TentativePFactory_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_ThresholdAFilterFactory.cpp;Utils/ExplicitInstantiation/MueLu_ToggleCoordinatesTransferFactory.cpp;Utils/ExplicitInstantiation/MueLu_TogglePFactory.cpp;Utils/ExplicitInstantiation/MueLu_TopRAPFactory.cpp;Utils/ExplicitInstantiation/MueLu_TopSmootherFactory.cpp;Utils/ExplicitInstantiation/MueLu_TpetraOperator.cpp;Utils/ExplicitInstantiation/MueLu_TransPFactory.cpp;Utils/ExplicitInstantiation/MueLu_TrilinosSmoother.cpp;Utils/ExplicitInstantiation/MueLu_UnsmooshFactory.cpp;Utils/ExplicitInstantiation/MueLu_UserPFactory.cpp;Utils/ExplicitInstantiation/MueLu_Utilities.cpp;Utils/ExplicitInstantiation/MueLu_UtilitiesBase.cpp;Utils/ExplicitInstantiation/MueLu_Utilities_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_VariableDofLaplacianFactory.cpp;Utils/ExplicitInstantiation/MueLu_SemiCoarsenPFactory.cpp;Utils/ExplicitInstantiation/MueLu_UzawaSmoother.cpp;Utils/ExplicitInstantiation/MueLu_VisualizationHelpers.cpp;Utils/ExplicitInstantiation/MueLu_ZoltanInterface.cpp;Utils/ExplicitInstantiation/MueLu_Zoltan2Interface.cpp;Utils/ExplicitInstantiation/MueLu_Aggregates.cpp;Utils/ExplicitInstantiation/MueLu_Aggregates_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase1Algorithm.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase1Algorithm_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase2aAlgorithm.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase2aAlgorithm_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase2bAlgorithm.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase2bAlgorithm_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase3Algorithm.cpp;Utils/ExplicitInstantiation/MueLu_AggregationPhase3Algorithm_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_AmalgamationInfo.cpp;Utils/ExplicitInstantiation/MueLu_CoupledAggregationCommHelper.cpp;Utils/ExplicitInstantiation/MueLu_CoupledAggregationFactory.cpp;Utils/ExplicitInstantiation/MueLu_Graph.cpp;Utils/ExplicitInstantiation/MueLu_IsolatedNodeAggregationAlgorithm.cpp;Utils/ExplicitInstantiation/MueLu_IsolatedNodeAggregationAlgorithm_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_IsorropiaInterface.cpp;Utils/ExplicitInstantiation/MueLu_LWGraph.cpp;Utils/ExplicitInstantiation/MueLu_LWGraph_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_LeftoverAggregationAlgorithm.cpp;Utils/ExplicitInstantiation/MueLu_LocalAggregationAlgorithm.cpp;Utils/ExplicitInstantiation/MueLu_OnePtAggregationAlgorithm.cpp;Utils/ExplicitInstantiation/MueLu_OnePtAggregationAlgorithm_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_PreserveDirichletAggregationAlgorithm.cpp;Utils/ExplicitInstantiation/MueLu_PreserveDirichletAggregationAlgorithm_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_RebalanceMapFactory.cpp;Utils/ExplicitInstantiation/MueLu_RepartitionInterface.cpp;Utils/ExplicitInstantiation/MueLu_UncoupledAggregationFactory.cpp;Utils/ExplicitInstantiation/MueLu_UncoupledAggregationFactory_kokkos.cpp;Utils/ExplicitInstantiation/MueLu_UserAggregationFactory.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Graph/MueLu_LinkedList.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListInterpreter.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListUtils.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_Describable.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_Factory.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_FactoryBase.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_Level.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_MasterList.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_NoFactory.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_VerboseObject.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/MueCentral/MueLu_VerbosityLevel.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Smoothers/MueLu_AmesosSmoother.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Smoothers/MueLu_IfpackSmoother.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/MueLu_Memory.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/MueLu_Monitor.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/MueLu_MutuallyExclusiveTime.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/MueLu_Utilities.cpp;/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Utils/ExplicitInstantiation/MueLu_FacadeClassBase.cpp
```

Error message:
```
[ 99%] Linking CXX executable MueLu_simple-factory-request-mechanism.exe
nvlink warning : Stack size for entry function '_ZN6Kokkos4Impl75_GLOBAL__N__51_tmpxft_0001554b_00000000_7_Kokkos_Cuda_Task_cpp1_ii_b2872e7123cuda_task_queue_executeEPNS0_9TaskQueueINS_4CudaEEEi' cannot be statically determined
../../src/libmuelu.a(MueLu_ML2MueLuParameterTranslator.cpp.o): In function `MueLu::ML2MueLuParameterTranslator::GetSmootherFactory(Teuchos::ParameterList const&, Teuchos::ParameterList&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.cpp:58: multiple definition of `MueLu::ML2MueLuParameterTranslator::GetSmootherFactory(Teuchos::ParameterList const&, Teuchos::ParameterList&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
../../src/Interface/libmuelu-interface.a(MueLu_ML2MueLuParameterTranslator.cpp.o):/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.cpp:58: first defined here
../../src/libmuelu.a(MueLu_ML2MueLuParameterTranslator.cpp.o): In function `MueLu::ML2MueLuParameterTranslator::SetParameterList(Teuchos::ParameterList const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.cpp:184: multiple definition of `MueLu::ML2MueLuParameterTranslator::SetParameterList(Teuchos::ParameterList const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
../../src/Interface/libmuelu-interface.a(MueLu_ML2MueLuParameterTranslator.cpp.o):/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ML2MueLuParameterTranslator.cpp:184: first defined here
../../src/libmuelu.a(MueLu_ParameterListInterpreter.cpp.o): In function `MueLu::LevenshteinDistance(char const*, unsigned long, char const*, unsigned long)':
/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListInterpreter.cpp:54: multiple definition of `MueLu::LevenshteinDistance(char const*, unsigned long, char const*, unsigned long)'
../../src/Interface/libmuelu-interface.a(MueLu_ParameterListInterpreter.cpp.o):/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListInterpreter.cpp:54: first defined here
../../src/libmuelu.a(MueLu_ParameterListUtils.cpp.o): In function `MueLu::GetMLSubList(Teuchos::ParameterList const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)':
/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListUtils.cpp:145: multiple definition of `MueLu::GetMLSubList(Teuchos::ParameterList const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)'
../../src/Interface/libmuelu-interface.a(MueLu_ParameterListUtils.cpp.o):/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListUtils.cpp:145: first defined here
../../src/libmuelu.a(MueLu_ParameterListUtils.cpp.o): In function `MueLu::replaceAll(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListUtils.cpp:175: multiple definition of `MueLu::replaceAll(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
../../src/Interface/libmuelu-interface.a(MueLu_ParameterListUtils.cpp.o):/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListUtils.cpp:175: first defined here
../../src/libmuelu.a(MueLu_ParameterListUtils.cpp.o): In function `MueLu::MergeParameterList(Teuchos::ParameterList const&, Teuchos::ParameterList&, bool)':
/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListUtils.cpp:58: multiple definition of `MueLu::MergeParameterList(Teuchos::ParameterList const&, Teuchos::ParameterList&, bool)'
../../src/Interface/libmuelu-interface.a(MueLu_ParameterListUtils.cpp.o):/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListUtils.cpp:58: first defined here
../../src/libmuelu.a(MueLu_ParameterListUtils.cpp.o): In function `MueLu::CreateSublists(Teuchos::ParameterList const&, Teuchos::ParameterList&)':
/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListUtils.cpp:65: multiple definition of `MueLu::CreateSublists(Teuchos::ParameterList const&, Teuchos::ParameterList&)'
../../src/Interface/libmuelu-interface.a(MueLu_ParameterListUtils.cpp.o):/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListUtils.cpp:65: first defined here
../../src/libmuelu.a(MueLu_ParameterListUtils.cpp.o): In function `MueLu::ExtractSetOfParameters(Teuchos::ParameterList const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListUtils.cpp:160: multiple definition of `MueLu::ExtractSetOfParameters(Teuchos::ParameterList const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
../../src/Interface/libmuelu-interface.a(MueLu_ParameterListUtils.cpp.o):/mnt/netapp/OHPC_home/crtrott/Trilinos/packages/muelu/src/Interface/MueLu_ParameterListUtils.cpp:160: first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [packages/muelu/test/factories/MueLu_simple-factory-request-mechanism.exe] Error 1
```

--- a/packages/muelu/src/CMakeLists.txt
+++ b/packages/muelu/src/CMakeLists.txt
@@ -416,9 +416,8 @@ INCLUDE_DIRECTORIES(${BDIR}/)
 APPEND_GLOB(HEADERS ${BDIR}/*.hpp)
 
 # Remove everything in the interface directory and add it to a list for muelu-interface instead
-# NOTE: Because of the funny way the glob mactros work, the use of two slashses for the APPEND_GLOB is critical
 SET(SOURCES_INTERFACE "")
-APPEND_GLOB(SOURCES_INTERFACE Interface//*.cpp)
+APPEND_GLOB(SOURCES_INTERFACE Interface/*.cpp)
 IF(${PACKAGE_NAME}_ENABLE_EXPLICIT_INSTANTIATION)
   APPEND_SET(SOURCES_INTERFACE ${DIR}/Utils/ExplicitInstantiation/MueLu_FacadeClassFactory.cpp)
   APPEND_SET(SOURCES_INTERFACE ${DIR}/Utils/ExplicitInstantiation/MueLu_ParameterListInterpreter.cpp)

